.github/workflows/pr-check.yaml

```diff
@@ -20,10 +20,65 @@ name: pr-check
 on: [pull_request]
 
 jobs:
-  pr-check-job:
+  
+  build:
+    name: Build
     strategy:
```

Comment:
It might interesting to: set a fail-fast on matrix build job, so we cancel other builds if one fails. To save some resources if the result is likely to be negative.
Then, I would use `needs: Build` in `Test` job, so we run all linters and tests just if we can actually build the changes. 
Take this suggestions as a follow up idea, if you like, Otherwise we are good.

Reply:
I'll do vice-versa. Test first and if it is good - build.